components/member_list_channel/member_list_channel.jsx

```diff
@@ -144,29 +217,31 @@ export default class MemberListChannel extends React.Component {
             usersToDisplay = [];
 
             for (let i = 0; i < users.length; i++) {
-                const user = users[i];
+                const user = Object.assign({}, users[i]);
+                user.value = user.id;
+                user.label = '@' + user.username;
+                users[i] = user;
```

Comment:
A bit of an early review, but there's a few things to note here:
1. We should never be mutating data that comes from the store. It should be treated as immutable even though JavaScript makes that hard to enforce it
2. On top of that, nothing in the state should be changed without setState since React uses that to know if should redraw parts of the app
3. The render function should never have any side effects (in this case, changing the state). It ideally should just take the state/props of the component and return whatever it renders as without touching or looking at anything else.

What you'll want to do here instead is, as you're filling up usersToDisplay, you'll also want to create a corresponding values array which has the corresponding IDs for each user in it, and then pass that to the MultiSelect